tests/class-mock-asset-registry.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Helper class used to mock script & style registries underlying wp_enqueue_*.
+ */
+
+namespace Asset_Loader\Tests;
+
+class Mock_Asset_Registry {
+	/**
+	 * Dictionary of registered objects in this registry.
+	 *
+	 * @var object[]
+	 */
+	public $registered;
+
+	/**
+	 * Dictionary of enqueued assets in this registry.
+	 *
+	 * @var string[]
+	 */
+	public $enqueued;
+
+	public function __construct() {
+		$this->reset();
```

Comment:
Why would you need to reset something on construction? Do you mis-use `__construct` somewhere in your code?

If not, I'd provide default values for the two properties/fields (and make them (`private`), and delete the constructor.